contracts/gateway/Gateway.sol

```diff
@@ -0,0 +1,202 @@
+
+        idFactory = IdFactory(idFactoryAddress);
+    }
+
+    /**
+     *  @dev Approve a signer to sign ONCHAINID deployments. If the Gateway is setup to require signature, only
+     *  deployments requested with a valid signature from an approved signer will be accepted.
+     *  If the gateway does not require a signature,
+     *  @param signer the signer address to approve.
+     */
+    function approveSigner(address signer) external onlyOwner {
+        if (signer == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (approvedSigners[signer]) {
+            revert SignerAlreadyApproved();
+        }
+
+        approvedSigners[signer] = true;
+
+        emit SignerApproved(signer);
+    }
+
+    /**
+     *  @dev Revoke a signer to sign ONCHAINID deployments.
+     *  @param signer the signer address to revoke.
+     */
+    function revokeSigner(address signer) external onlyOwner {
+        if (signer == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (!approvedSigners[signer]) {
+            revert SignerAlreadyNotApproved();
+        }
+
+        delete approvedSigners[signer];
+
+        emit SignerRevoked(signer);
+    }
+
+    /**
+     *  @dev Deploy an ONCHAINID using a factory. The operation must be signed by
+     *  an approved public key. This method allow to deploy an ONCHAINID using a custom salt.
+     *  @param identityOwner the address to set as a management key.
+     *  @param salt to use for the deployment.
+     *  @param signatureExpiry the block timestamp where the signature will expire.
+     *  @param signature the approval containing the salt and the identityOwner address.
+     */
+    function deployIdentityWithSalt(
+        address identityOwner,
+        string memory salt,
+        uint256 signatureExpiry,
+        bytes calldata signature
+    ) external returns (address) {
+        if (identityOwner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (signatureExpiry != 0 && signatureExpiry < block.timestamp) {
+            revert ExpiredSignature();
+        }
+
+        address signer = ECDSA.recover(
+            keccak256(
+                abi.encode(
+                    "Authorize ONCHAINID deployment",
+                    identityOwner,
+                    salt,
+                    signatureExpiry
+                )
+            ).toEthSignedMessageHash(),
+            signature
+        );
+
+        if (!approvedSigners[signer]) {
+            revert UnapprovedSigner();
+        }
+
+        if (revokedSignatures[signature]) {
+            revert RevokedSignature();
+        }
+
+        return idFactory.createIdentity(identityOwner, salt);
+    }
+
+    /**
+     *  @dev Deploy an ONCHAINID using a factory using the identityOwner address as salt.
+     *  @param identityOwner the address to set as a management key.
+     */
+    function deployIdentityForWallet(address identityOwner) external returns (address) {
```

Comment:
we should also have a function to deploy identities with multiple wallets as management key, which is a function that now exists on the Factory and that we use for our clients.

Reply:
that function should be protected though, and accessible at first only by Tokeny, maybe by some other actors after.

Reply:
Added a `function deployIdentityWithSaltAndManagementKeys(address identityOwner, string memory salt, bytes32[] calldata managementKeys, uint256 signatureExpiry, bytes calldata signature)` method in https://github.com/onchain-id/solidity/pull/89/commits/4d2d40d6ece2568b422dd9c355fce0dae08ab386 . This method requires a valid signature that contains the list of management keys to add on the identity.